sprint3/ProductApi.js

```diff
@@ -0,0 +1,373 @@
+    const {name,description, price, tags} = req.body;
+    if (!name|| !description ||!price || !tags){
+        const err = new Error("invalid body data");
+        err.status = 400;
+        return next(err)
+    }
+
+    try{
+        const Product = await prisma.product.create({
+            data:{
+                name,
+                description,
+                price,
+                tags,
+            }
+        });
+        console.log("post success");
+        return res.status(201).send(Product);
+        
+    }catch(error){
+        console.log('post product failed because of server error');
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});
+
+//게시글 수정하기
+ProductRouter.patch('/detail/:id/modify', async (req,res,next) =>{
+    const {name, description, price, tags} = req.body;
+    const id = Number(req.params.id) ;
+
+    if (!id){
+        const err = new Error("invalid parameter")
+        err.status = 400;
+        return next(err);
+    }
+
+    if (!name|| !description ||!price || !tags){
+        const err = new Error("invalid body data");
+        err.status = 400;
+        return next(err);
+    }
+
+    try{
+        const product= await prisma.product.update({
+            where: {id},
+            data: {
+                name,
+                description,
+                price,
+                tags
+            }
+        });
+        console.log("patch success");
+        return res.status(200).send(product);
+        
+    }catch(error){
+        console.log('patch product failed because of server error');
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});
+
+//게시글 삭제하기 
+ProductRouter.delete('/detail/:id', async (req,res,next) =>{
+    const id = Number(req.params.id) ;
+    if (!id){
+        const err = new Error("invalid parameter")
+        err.status = 400;
+        return next(err);
+    }
+
+    const product = await prisma.Product.findUnique({
+            where:{id}
+    });
+    if (!product){
+        const err = new Error("No content")
+        err.status = 404;
+        return next(err);
+    }
+
+    try{
+        await prisma.Product.delete({
+            where:{id}
+        });
+        console.log("deleting success");
+        return res.status(204).send("deleting successed");
+        
+    }catch(error){
+        console.log('deleting product failed because of server error');
+        const err = new Error("Server Error");
+        err.status = 500;
+        return next(err);
+    }
+});
+
+
+
+
+
+//------------------------------댓글-----------------
+
+
+
+//모든 댓글 보기
+ProductRouter.get('/comments', async (req,res,next) =>{
+    try{
+        let {take = '10',skip= '1',commentId = '1'} = req.query;
+        take = parseInt(take);
+        skip = parseInt(skip);
+        commentId = parseInt(commentId);
```

Comment:
이 쿼리에 대한 검증도 같이 해주세요.
```suggestion    
    let { sort = 'recent', skip = '0', take = '10', searchName, searchDescription } = req.query as any;
    const _skip = Number(skip); const _take = Number(take);
    const skipNum = Number.isFinite(_skip) && _skip >= 0 ? _skip : 0;
    const takeNum = Number.isFinite(_take) && _take > 0 ? Math.min(_take, 100) : 10;
```